core/src/main/java/io/questdb/griffin/NetworkSqlExecutionCircuitBreaker.java

```diff
@@ -28,42 +28,66 @@
 import io.questdb.network.NetworkFacade;
 import io.questdb.std.MemoryTag;
 import io.questdb.std.Unsafe;
+import io.questdb.std.datetime.microtime.MicrosecondClock;
 
 import java.io.Closeable;
 
-public class HttpSqlExecutionInterruptor implements SqlExecutionInterruptor, Closeable {
+public class NetworkSqlExecutionCircuitBreaker implements SqlExecutionCircuitBreaker, Closeable {
     private final NetworkFacade nf;
-    private final int nIterationsPerCheck;
+    private final int throttle;
     private final int bufferSize;
     private long buffer;
-    private int nIterationsSinceCheck;
+    private int testCount;
     private long fd = -1;
+    private long powerDownDeadline;
+    private final MicrosecondClock clock;
+    private final long maxTime;
 
-    public HttpSqlExecutionInterruptor(SqlInterruptorConfiguration configuration) {
+    public NetworkSqlExecutionCircuitBreaker(SqlExecutionCircuitBreakerConfiguration configuration) {
         this.nf = configuration.getNetworkFacade();
-        this.nIterationsPerCheck = configuration.getCountOfIterationsPerCheck();
+        this.throttle = configuration.getCircuitBreakerThrottle();
         this.bufferSize = configuration.getBufferSize();
-        buffer = Unsafe.malloc(bufferSize, MemoryTag.NATIVE_HTTP_CONN);
+        this.buffer = Unsafe.malloc(bufferSize, MemoryTag.NATIVE_DEFAULT);
+        this.clock = configuration.getClock();
+        this.maxTime = configuration.getMaxTime();
     }
 
     @Override
-    public void checkInterrupted() {
-        assert fd != -1;
-        if (nIterationsSinceCheck == nIterationsPerCheck) {
-            nIterationsSinceCheck = 0;
-            checkConnection();
+    public void powerUp() {
+        final long ticks = clock.getTicks();
+        // test for overflow
+        if ((maxTime > 0) && (ticks > Long.MAX_VALUE - maxTime)) {
```

Comment:
Is it year 10,000 protection? Micro from epoch does not going to overflow 64bit in next few hundred years

Reply:
the default `maxTime` value is `Long.MAX_VALUE`. This is a protection from `maxValue` being set too high